### ISSUE TITLE
508-defect-2 [SCREEN READER]: Notes should be programatically linked with aria-describedby to their relevant fields

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#22303]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a screen reader user, I want to hear any relevant notes to a field that a sighted user would be able to read.

## Details

Screen reader users rely on `tab` to navigate through forms in what is called "forms mode." As such, any elements that are not focusable will likely be completely skipped unless programmatically linked to a field or focusable element.

That being said, the notes provided above certain fields like date of birth will likely be skipped over by screen reader users.

## Acceptance Criteria

- [ ] When a user tabs into a field with notes, the notes are read out
  - [ ] Step 3 of 6: Provide your personal information, birth, sex, and SSN fields
  - [ ] Step 4 of 6: Provide your contact information, email address and phone number fields
  - [ ] Fill out this form, birth and SSN fields

## Proposed Solution 

Use `aria-describedby` to connect the note to the field

### Current

```html
<div class="schemaform-field-template">
  <label id="root_ssn-label" class="schemaform-label" for="root_ssn">
    Social Security number (SSN)
  </label>
  <span>
    <b>Note: </b>Your <abbr title="Social Security Number">SSN</abbr> helps us match your information to your Veteran records. We can then share your vaccine plans with your local VA health facility so they can contact you when you’re eligible to get a vaccine.
  </span>
  <div class="schemaform-widget-wrapper">
    <input type="text" id="root_ssn" name="root_ssn" class="usa-input-medium" value="">
  </div>
<div>
```

### Proposed
```html
<div class="schemaform-field-template">
  <label id="root_ssn-label" class="schemaform-label" for="root_ssn">
    Social Security number (SSN)
  </label>
  <!-- Convert this to a p and add a unique id -->
  <p id="ssn-note">
    <!-- Use strong, not b -->
    <strong>Note: </strong>Your <abbr title="Social Security Number">SSN</abbr> helps us match your information to your Veteran records. We can then share your vaccine plans with your local VA health facility so they can contact you when you’re eligible to get a vaccine.
  </p>
  <div class="schemaform-widget-wrapper">
    <!-- Link the input to the note using aria-describedby -->
    <input aria-describedby="ssn-note" type="text" id="root_ssn" name="root_ssn" class="usa-input-medium" value="">
  </div>
<div>
```

## Screenshots or Trace Logs
<img width="1118" alt="Screen Shot 2021-03-29 at 11 53 32 AM" src="https://user-images.githubusercontent.com/14154792/112870087-d4cd2480-908b-11eb-8a4c-6ebffc112a50.png">
<img width="1274" alt="Screen Shot 2021-03-29 at 12 48 20 PM" src="https://user-images.githubusercontent.com/14154792/112871387-380b8680-908d-11eb-9ab0-8d87ed63f3b8.png">
<img width="762" alt="Screen Shot 2021-03-29 at 3 30 47 PM" src="https://user-images.githubusercontent.com/14154792/112889580-d30f5b00-90a3-11eb-9d79-e47b31143312.png">
